src/js/BookReader/ModeThumb.js

```diff
@@ -0,0 +1,364 @@
+            event.stopPropagation();
+          });
+
+          this.br.refs.$brPageViewEl.append($pageContainer);
+          imagesToDisplay.push(leaf);
+
+          /* get thumbnail's reducer */
+          const idealReduce = floor(book.getPageWidth(leaf) / this.br.thumbWidth);
+          const nearestFactor2 = 2 * Math.round(idealReduce / 2);
+          const thumbReduce = nearestFactor2;
+
+          const baseCSS = { width: `${leafWidth}px`, height: `${leafHeight}px` };
+          if (this.br.imageCache.imageLoaded(leaf, thumbReduce)) {
+          // send to page
+            $pageContainer.append($(this.br.imageCache.image(leaf, thumbReduce)).css(baseCSS));
+          } else {
+          // lazy load
+            const $lazyLoadImgPlaceholder = document.createElement('img');
+            $($lazyLoadImgPlaceholder)
+              .attr('src', `${this.br.imagesBaseURL}transparent.png`)
+              .css(baseCSS)
+              .addClass('BRlazyload')
+            // Store the leaf/index number to reference on url swap:
+              .attr('data-leaf', leaf)
+              .attr('data-reduce', thumbReduce)
+              .attr('data-row', row)
+              .attr('alt', 'Loading book image');
+            $pageContainer.append($lazyLoadImgPlaceholder);
+          }
+        }
+      }
+    }
+
+    // Remove thumbnails that are not to be displayed
+    for (const row of this.br.displayedRows) {
+      if (notInArray(row, rowsToDisplay)) {
+        for (const { num: index } of leafMap[row]?.leafs) {
+          if (!imagesToDisplay?.includes(index)) {
+            this.br.$(`.pagediv${index}`)?.remove();
+          }
+        }
+      }
+    }
+
+    // Update which page is considered current to make sure a visible page is the current one
+    const currentIndex = this.br.currentIndex();
+    if (currentIndex < leastVisible) {
+      this.br.updateFirstIndex(leastVisible);
+    } else if (currentIndex > mostVisible) {
+      this.br.updateFirstIndex(mostVisible);
+    }
+
+    // remember what rows are displayed
+    this.br.displayedRows = rowsToDisplay.slice();
+
+    // remove previous highlights
+    this.br.$('.BRpagedivthumb_highlight').removeClass('BRpagedivthumb_highlight');
+
+    // highlight current page
+    this.br.$('.pagediv' + this.br.currentIndex()).addClass('BRpagedivthumb_highlight');
+
+    this.br.lazyLoadThumbnails();
+
+    this.br.updateToolbarZoom(this.br.reduce);
+  }
+
+  lazyLoadThumbnails() {
+    const batchImageRequestByRow = async ($images) => {
+      await sleep(300);
+      $images.each((index, $imgPlaceholder) => this.br.lazyLoadImage($imgPlaceholder));
+    };
+
+    this.br.displayedRows.forEach((row) => {
+      const $imagesInRow = this.br.refs.$brPageViewEl.find(`[data-row="${row}"]`);
+      batchImageRequestByRow($imagesInRow);
+    });
+  }
+
+  /**
+   * Replaces placeholder image with real one
+   *
+   * @param {$lazyLoadImgPlaceholder} imgPlaceholder
+   */
+  lazyLoadImage(imgPlaceholder) {
```

Comment:
I'm going to move this back in a future PR when I update Mode1Up to also use the imagecache.